sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -240,7 +255,11 @@ public void startBundle(StartBundleContext context) {
     public void processElement(ProcessContext context) {
       Object[] avaticaRow = rowToAvatica(context.element());
       if (avaticaRow.length == 1) {
```

Comment:
It seems like this could be solved in a much simpler manner if you removed the unboxing (this `if` block) from `processElement` and instead did it in collect. So collect becomes:

```
if  (avaticaRow.length == 1) {
   List<Object> unboxed = new ArrayList();
   // code to iterate and unbox values
   return Linq4j.asEnumerable(unboxed);
}
return Linq4j.asEnumerable(values);
```

Reply:
I think NPE will still appear. NPE happens when add a null value in `ConcurrentLinkedQueue`. So the resolution seems to me is to walk around on `ConcurrentLinkedQueue`

Reply:
A `Object[]` containing a single null element isn't null itself, so you can put that in the queue.

Reply:
I think that things like `Object[]` with a single `null` element are very hard to interpret. Special class is self documenting in this case.

Reply:
@apilloud 

My understanding now is if a `Object[]` contains one single element, that single element has to be added to queue (not Object[]). That's why NPE happens.

I don't think there will be a `null` for `Object[]`.

Reply:
Thanks all. 

I have updated this PR to what @apilloud suggested, which is add `Object[]` to queue no matter if the `Object[]` contains one or more elements, and unbox it in `collect` functions.